pkg/daemon/update.go

```diff
@@ -810,6 +815,22 @@ func (dn *Daemon) updateOS(config *mcfgv1.MachineConfig) error {
 	return nil
 }
 
+func (dn *Daemon) writeProxyEnvs() error {
+	proxyCfg, err := dn.proxyLister.Get("cluster")
+	if err != nil {
+		if !apierrors.IsNotFound(err) {
+			return err
+		}
+		// no proxy configured
+		return os.Remove(proxyPath)
+	}
+ 	proxyEnvs := fmt.Sprintf(`HTTP_PROXY=%s
+HTTPS_PROXY=%s
+NO_PROXY=%s
+`, proxyCfg.Status.HTTPProxy, proxyCfg.Status.HTTPSProxy, proxyCfg.Status.NoProxy)
```

Comment:
How do we distinguish between user has set the proxy resource spec but whatever operator has not reconciled and set the status yet, and empty values for each? At the least, there could be an unnecessary flip-flop in this file

Reply:
yeah, this isn't hashed out though, just a preliminary test - we might want to go https://github.com/openshift/machine-config-operator/pull/883#issuecomment-504977641 anyway

Reply:
> How do we distinguish between user has set the proxy resource spec but whatever operator has not reconciled and set the status yet

This could be a reason for sycing proxies specifically (like we do for ImageConfig and ContainerRuntimeConfig). On the other end, since this rerun on failure (assuming your connections have to go through a proxy) we can fail this and once Status is again populated, this will go through. But I'm starting to prefer the first option. Emtpy values go through systemd env file and process as always (unset?)

Reply:
note to self, we need to check `status != nil` as per slack conversations